File: TodoList_App/todolist_app/lib/domain/entities/todo_entity.dart
import '../enum/status_todo_enum.dart';

class TodoEntity {
  TodoEntity({
    this.todoId,
    required this.title,
    required this.statusTodo,
    required this.description,
    required this.dhInicio,
  });

  String title;
  String? todoId;
  String description;
  StatusTodoEnum statusTodo;
  DateTime dhInicio;

  factory TodoEntity.fromJson(Map<String, dynamic> json) => TodoEntity(
        statusTodo: StatusTodoEnum.values[json["statusTodo"]],
        description: json["description"] as String,
        todoId: json["todoId"] as String,
        title: json["title"] as String,
        dhInicio: json["dhInicio"] as DateTime,
      );

  Map<String, dynamic> toJson() => {
        "statusTodo": statusTodo.index,
        "description": description,
        "title": title,
        "dhInicio": dhInicio.toIso8601String()
      };
}

File: WebAPI_TodoList/WebAPI_TodoList.Domain/Entities/TodoEntity.cs
﻿using WebAPI_TodoList.Infra.Data.Identity;
using WebAPI_TodoList.Domain.Enum;

namespace WebAPI_TodoList.Domain.Entities;

public sealed class TodoEntity
{
    public Guid TodoId { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public StatusTodoEnum StatusTodo { get; set; }
    public DateTime DhInicio { get; set; }

    public string UserId { get; set; }
    public UserCustomEntity UserCustom { get; set; }

}
